staging/src/k8s.io/apiserver/pkg/endpoints/filters/request_deadline.go

```diff
@@ -147,6 +147,29 @@ func failedErrorHandler(s runtime.NegotiatedSerializer, statusError *apierrors.S
 	})
 }
 
+// RequestContextWithUpperBound returns a new deadline bound context for the request.
+//
+// If the request context is already setup with a deadline then use the
+// requestTimeoutUpperBound as the upper bound.
+// If the request context is not setup with a deadline then use:
+//  - user specified timeout in the request URI, otherwise
+//  - use the default value in requestTimeoutUpperBound
+func RequestContextWithUpperBound(req *http.Request, requestTimeoutUpperBound time.Duration) (context.Context, context.CancelFunc) {
```

Comment:
am not happy with the func. The `requestTimeoutUpperBound` is not an upper bound in all cases, but can be overridden by the request timeout if the context didn't have a deadline yet. So in the latter case it is a default, not an upper bound.